shell-manager/src/gadget/native.rs

```diff
@@ -117,17 +125,39 @@ async fn handle_github_source(
 
         for blueprint in blueprints {
             if blueprint.blueprint_id == blueprint_id {
-                for svc in &blueprint.services {
-                    let service_id = svc.id;
+                for service_id in &blueprint.services {
                     let sub_service_str = format!("{service_str}-{service_id}");
                     // Each spawned binary will effectively run a single "RoleType" in old parlance
                     let arguments = utils::generate_process_arguments(
                         shell_config,
                         shell_manager_opts,
                         blueprint_id,
-                        service_id,
+                        *service_id,
                     )?;
 
+                    let env_vars = if blueprint.registration_mode {
+                        let mut env_vars = std::env::vars().collect::<Vec<_>>();
+                        // RPC_URL: The remote RPC url for tangle
+                        // KEYSTORE_URI: the keystore file where the keys are stored and could be retrieved.
+                        // DATA_DIR: is an isolated path for where this gadget should store its data (database, secrets, …etc)
+                        // BLUEPRINT_ID: the active blueprint ID for this gadget
+                        // REGISTRATION_MODE_ON set to any value.
+                        env_vars.push(("RPC_URL".to_string(), shell_config.url.to_string()));
+                        env_vars.push((
+                            "KEYSTORE_URI".to_string(),
+                            "file://./target/keystore".to_string(),
+                        ));
```

Comment:
That also should come from the CLI.

Reply:
Would you prefer the shell/blueprint manager config, or, the individual shell config?

Reply:
It should be in the manager config nonetheless.